ray-operator/apis/ray/v1alpha1/raycluster_types.go

```diff
@@ -30,9 +30,8 @@ type HeadGroupSpec struct {
 	ServiceType v1.ServiceType `json:"serviceType"`
 	// EnableIngress indicates whether operator should create ingress object for head service or not.
 	EnableIngress *bool `json:"enableIngress,omitempty"`
-	// Number of desired pods in this pod group. This is a pointer to distinguish between explicit
-	// zero and not specified. Defaults to 1.
-	Replicas *int32 `json:"replicas"`
+	// HeadGroupSpec.Replicas is deprecated and ignored; there can only be one head pod per Ray cluster.
+	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
We need to make sure all the references are cleaned up. I remember some logics do use it and we can change to use const 1 instead.  Please search the references and have a check.

Reply:
I actually don't see any references.
The reconcilePods logic definitely doesn't use it -- it just enforces one head.
Grepping for `\.Replicas` indicates it's only accessed on workers (in `util.go` and `raycluster_controller.go`).
vim-go's `:GoReferrers` also doesn't turn up anything (outside of test files)

Reply:
@Jeffwan do you think you could double-check this?

Reply:
Let me take a look. If there's nothing left, we can merge the PR.